tests/test_common.py

```diff
@@ -0,0 +1,251 @@
+import dataclasses
+
+import pytest
+
+from genieutils.common import GenieClass, ByteHandler
+from genieutils.versions import Version
+
+class TestGenieClass:
+    @pytest.fixture
+    def genie_class(self):
+        return GenieClass()
+    
+    def test_from_bytes(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.from_bytes(ByteHandler(b''))
+
+    def test_from_bytes_with_count(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.from_bytes_with_count(ByteHandler(b''), 0)
+
+    def test_to_bytes(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.to_bytes(Version.UNDEFINED)
+
+    def test_write_debug_string(self, genie_class: GenieClass):
+        assert genie_class.write_debug_string("foobar") == b'`\n\x06\x00foobar'
+
+    def test_write_string(self, genie_class: GenieClass):
+        assert genie_class.write_string(6, "foobar") == b'foobar'
+
+    def test_write_int_8(self, genie_class: GenieClass):
+        assert genie_class.write_int_8(10) == b'\n'
```

Comment:
I don't think it's obvious that `\n` is `10`. Maybe use `\x0a` instead? In the other tests, too.

Reply:
Fair enough. Though if I address the first comment then this one becomes obsolete I believe